src/components/admin/InviteAdminModal/InviteAdminModal.tsx

```diff
@@ -27,29 +46,91 @@ export const InviteAdminModal: React.FC<InviteAdminModalProps> = ({
     []
   );
 
-  const { register } = useForm({
+  const { register, watch } = useForm({
     reValidateMode: "onChange",
     defaultValues,
   });
 
-  const addAdmin = async () => {
+  const searchQuery = watch("userName");
+
+  const { value: algoliaValue } = useAlgoliaSearch(searchQuery);
+
+  const foundUsers = algoliaValue?.[AlgoliaSearchIndex.USERS].hits;
```

Comment:
Not sure about the value of `AlgoliaSearchIndex.USERS` but since there are null/undefined guards already...
```suggestion
  const foundUsers = algoliaValue?.[AlgoliaSearchIndex.USERS]?.hits;
```